llvm/tools/sycl-post-link/CompileTimePropertiesPass.cpp

```diff
@@ -0,0 +1,152 @@
+//===---- CompileTimePropertiesPass.cpp - SYCL Compile Time Props Pass ----===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// See comments in the header.
+//===----------------------------------------------------------------------===//
+
+#include "CompileTimePropertiesPass.h"
+#include "DeviceGlobals.h"
+
+#include "llvm/ADT/APInt.h"
+#include "llvm/ADT/StringMap.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/IR/Module.h"
+
+using namespace llvm;
+
+namespace {
+
+constexpr StringRef SYCL_HOST_ACCESS_ATTR = "host_access";
+
+constexpr StringRef SPIRV_DECOR_MD_KIND = "spirv.Decorations";
+// The corresponding SPIR-V OpCode for the host_access property is documented
+// in the SPV_INTEL_global_variable_decorations design document:
+// https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/DeviceGlobal/SPV_INTEL_global_variable_decorations.asciidoc#decoration
+constexpr uint32_t SPIRV_HOST_ACCESS_DECOR = 6147;
+constexpr uint32_t SPIRV_HOST_ACCESS_DEFAULT_VALUE = 2; // Read/Write
+
+enum class DecorValueTy {
+  // the value is an unsigned number (uint32_t)
```

Comment:
I think that these comments are not necessary.

Reply:
I'm agree, they have been removed. Thank you.